pertpy/tools/_differential_gene_expression/_simple_tests.py

```diff
@@ -144,19 +171,123 @@ class WilcoxonTest(SimpleComparisonBase):
     """
 
     @staticmethod
-    def _test(x0: np.ndarray, x1: np.ndarray, paired: bool, **kwargs) -> float:
+    def _test(x0: np.ndarray, x1: np.ndarray, paired: bool, **kwargs) -> dict[str, float]:
+        """Perform an unpaired or paired Wilcoxon/Mann-Whitney-U test."""
+
         if paired:
-            return scipy.stats.wilcoxon(x0, x1, **kwargs).pvalue
+            test_result = scipy.stats.wilcoxon(x0, x1, **kwargs)
         else:
-            return scipy.stats.mannwhitneyu(x0, x1, **kwargs).pvalue
+            test_result = scipy.stats.mannwhitneyu(x0, x1, **kwargs)
+
+        return {
+            "p_value": test_result.pvalue,
+            "statistic": test_result.statistic,
+        }
 
 
 class TTest(SimpleComparisonBase):
-    """Perform a unpaired or paired T-test"""
+    """Perform a unpaired or paired T-test."""
 
     @staticmethod
-    def _test(x0: np.ndarray, x1: np.ndarray, paired: bool, **kwargs) -> float:
+    def _test(x0: np.ndarray, x1: np.ndarray, paired: bool, **kwargs) -> dict[str, float]:
         if paired:
-            return scipy.stats.ttest_rel(x0, x1, **kwargs).pvalue
+            test_result = scipy.stats.ttest_rel(x0, x1, **kwargs)
         else:
-            return scipy.stats.ttest_ind(x0, x1, **kwargs).pvalue
+            test_result = scipy.stats.ttest_ind(x0, x1, **kwargs)
+
+        return {
+            "p_value": test_result.pvalue,
+            "statistic": test_result.statistic,
+        }
+
+
+class PermutationTest(SimpleComparisonBase):
+    """Perform a permutation test.
+
+    The permutation test relies on another test statistic (e.g. t-statistic or your own) to obtain a p-value through
+    random permutations of the data and repeated generation of the test statistic.
+
+    For paired tests, each paired observation is permuted together and distributed randomly between the two groups. For
+    unpaired tests, all observations are permuted independently.
+
+    The null hypothesis for the unpaired test is that all observations come from the same underlying distribution and
+    have been randomly assigned to one of the samples.
+
+    The null hypothesis for the paired permutation test is that the observations within each pair are drawn from the
+    same underlying distribution and that their assignment to a sample is random.
+    """
+
+    @staticmethod
+    def _test(
+        x0: np.ndarray,
+        x1: np.ndarray,
+        paired: bool,
+        test_statistic: type["SimpleComparisonBase"] | Callable = WilcoxonTest,
+        n_permutations: int = 1000,
+        **kwargs,
+    ) -> dict[str, float]:
+        """Perform a permutation test.
+
+        This function relies on another test (e.g. WilcoxonTest) to generate a test statistic for each permutation.
+
+        .. code-block:: python
```

Comment:
Pretty sure that we're not using codeblocks but a different syntax for the examples. Could you please harmonize it here? I also think that the examples always come last in the docstring.